embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryCacheManager.kt

```diff
@@ -62,9 +62,14 @@ internal class EmbraceDeliveryCacheManager(
             val sessionId = sessionMessage.session.sessionId
             val writeSync = snapshotType == SessionSnapshotType.JVM_CRASH
             val snapshot = snapshotType == SessionSnapshotType.PERIODIC_CACHE
+
             saveBytes(sessionId, writeSync, snapshot) { filename: String ->
                 Systrace.traceSynchronous("serialize-session") {
-                    cacheService.writeSession(filename, sessionMessage)
+                    if (cachedSessions.containsKey(sessionId)) {
```

Comment:
does this require synchronization? how do we ensure that there isn't a race between the check if something is in the cache and then writing  to the cache?

Reply:
Good call. Will do